### PR TITLE
Improve service locator tests

### PR DESCRIPTION
**Pull request by portellaa**
* Add tests for more Unregister cases

### CODE REVIEW

File: AlicerceTests/Utils/ServiceLocatorTests.swift [hunk 1]
```diff
@@ -646,6 +669,30 @@ class ServiceLocatorTests : XCTestCase {
...
+        let testLazyServiceInt: () -> Int = {
+            return 1
+        }
+
+        do {
+            try serviceLocator.register(name: serviceName, testLazyServiceInt)
+
+            try serviceLocator.unregister(Double.self, name: serviceName)
+
+            let _: Int = try serviceLocator.get(name: serviceName)
```
Comment (p4checo) [inline AlicerceTests/Utils/ServiceLocatorTests.swift]: This line isn't necessary 😜

File: AlicerceTests/Utils/ServiceLocatorTests.swift [hunk 2]
```diff
@@ -570,6 +570,29 @@ class ServiceLocatorTests : XCTestCase {
         }
     }
 
+    func testUnregister_UsingTypeInferenceAndName_WhenUnregisteringLazyService_ShouldRemoveService() {
+
+        let serviceName = "IntService"
+        let testLazyServiceInt: () -> (Int) = {
```
Comment (jakewelton) [inline AlicerceTests/Utils/ServiceLocatorTests.swift]: I don't think you need the brackets around `Int` :smile:

Comment (portellaa) [inline AlicerceTests/Utils/ServiceLocatorTests.swift]: no i don't 😂

**Comment by portellaa** [comment]
@p4checo @jakewelton fixed

**Comment by p4checo** [comment]
I think we are gathering coverage on the Unit Tests themselves 😂 

We have to make codecov.io ignore the test target's coverage 😆

**Comment by portellaa** [comment]
I was seeing that. this is stupid 🙄

**Pull request merged**